lib/ClangImporter/ImportDecl.cpp

```diff
@@ -7024,11 +7031,88 @@ ClangImporter::Implementation::createConstant(Identifier name, DeclContext *dc,
+                                              bool isFunction,
                                               ClangNode ClangN) {
   auto expr = new (SwiftContext) StringLiteralExpr(value, SourceRange());
-  return createConstant(name, dc, type, expr, convertKind, isStatic, ClangN);
+  if (isFunction)
+    return createFunction(name, dc, type, expr, convertKind, isStatic, ClangN);
+  else
+    return createConstant(name, dc, type, expr, convertKind, isStatic, ClangN);
 }
 
+ValueDecl *
+ClangImporter::Implementation::createFunction(Identifier name, DeclContext *dc,
+                                              Type type, Expr *valueExpr,
+                                              ConstantConvertKind convertKind,
+                                              bool isStatic,
+                                              ClangNode ClangN) {
+  auto &C = SwiftContext;
+
+  // Form the parameters.
+  SmallVector<ParameterList*, 3> parameters;
+  // The first parameter is 'self'.
+  if (dc->isTypeContext()) {
+    auto *selfDecl = ParamDecl::createSelf(SourceLoc(), dc, isStatic);
+    parameters.push_back(ParameterList::createWithoutLoc(selfDecl));
+  }
+  // The second is an empty tuple.
+  parameters.push_back(ParameterList::createEmpty(C));
+
+  // Form the type of the function.
+  auto funcType = ParameterList::getFullInterfaceType(type, parameters, C);
+
+  // Create the getter function declaration.
+  auto func = FuncDecl::create(
+      C, /*StaticLoc=*/SourceLoc(), StaticSpellingKind::None,
+      /*FuncLoc=*/SourceLoc(), /*Name=*/DeclName(C, name, parameters[0]),
+      /*NameLoc=*/SourceLoc(), /*Throws=*/false, /*ThrowsLoc=*/SourceLoc(),
+      /*AccessorKeywordLoc=*/SourceLoc(), /*GenericParams=*/nullptr, parameters,
+      /*TypeLoc=*/TypeLoc::withoutLoc(type), dc);
+  func->setStatic(isStatic);
+  func->setInterfaceType(funcType);
+  func->setAccessibility(getOverridableAccessibility(dc));
+
+  auto expr = valueExpr;
+
+  // If we need a conversion, add one now.
+  switch (convertKind) {
+  case ConstantConvertKind::None:
+    break;
+
+  case ConstantConvertKind::Construction:
+  case ConstantConvertKind::ConstructionWithUnwrap: {
+    auto typeRef = TypeExpr::createImplicit(type, C);
+
+    expr = CallExpr::createImplicit(C, typeRef, { expr }, { C.Id_rawValue });
+    if (convertKind == ConstantConvertKind::ConstructionWithUnwrap)
+      expr = new (C) ForceValueExpr(expr, SourceLoc());
+    break;
+  }
+
+  case ConstantConvertKind::Coerce:
+    break;
+
+  case ConstantConvertKind::Downcast: {
+    expr = new (C) ForcedCheckedCastExpr(expr, SourceLoc(), SourceLoc(),
+                                         TypeLoc::withoutLoc(type));
+    expr->setImplicit();
+    break;
+  }
+  }
+
+  // Create the return statement.
+  auto ret = new (C) ReturnStmt(SourceLoc(), expr);
+
+  // Finally, set the body.
+  func->setBody(BraceStmt::create(C, SourceLoc(),
+                                  ASTNode(ret),
```

Comment:
Nit: This should implicitly convert to `ASTNode`.  I think you want to make the array explicit

```cxx
func->setBody(BraceStmt::create(C, SourceLoc(), { ret }, SourceLoc()));
```